homeassistant/components/zha/config_flow.py

```diff
@@ -551,6 +551,40 @@ async def async_step_user(
 
         return await self.async_step_choose_serial_port(user_input)
 
+    async def async_step_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Confirm a discovery."""
+        self._set_confirm_only()
+
+        # Don't permit discovery if ZHA is already set up
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        # Without confirmation, discovery can automatically progress into parts of the
+        # config flow logic that interacts with hardware!
+        if user_input is not None or not onboarding.async_is_onboarded(self.hass):
+            # One last time, check if ZHA is already configured before touching hardware
+            if self._async_current_entries():
```

Comment:
This should already be covered by the above check.

Reply:
You're right. Existing unit test for this scenario passes without the check.